File: Gestor_canciones/src/gestorAplicación/ElementosLibreria/Biblioteca.java
package ElementosLibreria;
import java.util.ArrayList;
import Personas.Usuario;

public class Biblioteca {
    private Usuario usuario;
    private Cancion cancion_actual;
    private String estado_cancion;

    public Usuario getUsuario() {
        return usuario;
    }
    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }
    public Cancion getCancion_actual() {
        return cancion_actual;
    }
    public void setCancion_actual(Cancion cancion_actual) {
        this.cancion_actual = cancion_actual;
    }
    public String getEstado_cancion() {
        return estado_cancion;
    }
    public void setEstado_cancion(String estado_cancion) {
        this.estado_cancion = estado_cancion;
    }

    public Biblioteca( Usuario usuario){
        this.usuario = usuario;
    }

    public Mis_artistas acceder_misartistas(){
        return this.usuario.getMis_artistas();
    }

    public Playlist acceder_playlist(Playlist playlist){
        for(Playlist c: this.usuario.getPlaylists()) {
            if(c.equals(playlist)){
                return playlist;
            }
        }
        return playlist; //aquí hay que hacer lo de manejo de exepciones 
    }

}

File: Phyton/Gestor_canciones/src/gestorAplicacion/elementosLibreria/Biblioteca.py
from random import random
from gestorAplicacion.elementosLibreria.Genero import Genero
from gestorAplicacion.elementosLibreria.Cancion import Cancion
class Biblioteca:
    bibliotecas = []
    def __init__(self, usuario):
        self._usuario=usuario
        Biblioteca.bibliotecas.append(self)
        
    def getBibliotecas(self):
        return Biblioteca.bibliotecas
    def setBibliotecas(self, bibliotecas):
        Biblioteca.bibliotecas = bibliotecas
    
    def getUsuario(self):
        return self._usuario
    def setUsuario(self, usuario):
        self._usuario = usuario
        
    def acceder_misartistas(self):
        return self._usuario.getMis_artistas()
    
    
    
    def repo_aleatoria(self, playlist):
        if playlist in self._usuario.getPlaylists():
            index = int(random()* len(playlist.getCanciones()))
            return playlist.getCanciones()[index].play()
        else:
            return "La playlist "+ playlist.getNombre()+"no existe en el usuario "+self._usuario.getNombre()
    
    		